src/server/pkg/worker/api_server.go

```diff
@@ -490,6 +490,9 @@ func HashDatum(pipelineInfo *pps.PipelineInfo, data []*Input) (string, error) {
 		hash.Write(datum.FileInfo.Hash)
 	}
 
+	// We set env to nil because if env contains more than one elements,
+	// since it's a map, the output of Marshal() can be non-deterministic.
+	pipelineInfo.Transform.Env = nil
```

Comment:
I see, would you be opposed to doing it anyway? Even though it's not necessary now I feel like it's possible that at a later point someone could use `HashDatum` and not realize that it's modifying the `PipelineInfo` it gets passed and wind up with a pretty confusing bug. Alternatively I guess we could just leave a comment, but since it doesn't require much code to actually fix eliminate the issue that seems like the easiest path to me.